src/api.c

```diff
@@ -1258,6 +1258,7 @@ yaml_document_add_sequence(yaml_document_t *document,
     yaml_node_t node;
 
     assert(document);   /* Non-NULL document object is expected. */
+    if (STACK_NULL(&context, document->nodes)) goto error;
```

Comment:
I guess same check would have to be added to yaml_document_add_scalar, yaml_document_add_mapping etc. to make this useful.

Reply:
Yes, there are other places needs stack checking. I saw your discussions and not sure this cve needs fix or not, just sending this patch to discuss. Will send a V2.